### PR TITLE
Replacing global themed sass-vars to css-vars

### PR DESCRIPTION
**Pull request by ixrock**
**PR highlights:**

- Having default css-variables in app from generated default dark theme `lens-dark.json`
- Better IDE/`vscode` css-variables support / auto-completion
- More consistent styles overall
- Possibility to get rid of `sass` in future at all (if `postcss` would be just enough)

Screenshot 2021-11-11 at 20 57 42

**Recommended `vscode` extension for css-variables auto-complete:**
https://marketplace.visualstudio.com/items?itemName=phoenisx.cssvar

And add this to `settings.json` for auto-scanning css-variables in the project from all possible files:

```json
    "cssvar.files": [
        "./src/renderer/**/*.css",
        "./src/renderer/**/*.scss",
    ],
```

### CODE REVIEW

File: src/renderer/themes/theme-vars.css
Comment (Nokel81) [inline src/renderer/themes/theme-vars.css:1]: I don't think this file should be committed, should it?

File: build/build_theme_vars.ts
Comment (Nokel81) [inline build/build_theme_vars.ts:24]: Why only dark?

File: src/renderer/themes/theme-vars.css
Comment (ixrock) [reply src/renderer/themes/theme-vars.css:1]: Yes, it should. It's used in app as default values for themed variables + used by editor's css-vars auto-completion.

_P.S._ Most `vscode` plugins for autocompleting css-variables support only defining exact file to read for variables, so could be useful too.

File: build/build_theme_vars.ts
Comment (ixrock) [reply build/build_theme_vars.ts:24]: Only? Because default could be just one as name says `defaultBaseLensTheme` _(could be default white tought)_

Comment (Nokel81) [reply build/build_theme_vars.ts:24]: Ah so this is only for the auto-complete

Comment (jim-docker) [reply build/build_theme_vars.ts:24]: There seem to be more colors in `lens-light.json`

**Review by jim-docker — COMMENTED** [review]
I'm seeing this:

Screen Shot 2021-11-12 at 12 54 24 PM

should be:

Screen Shot 2021-11-12 at 12 40 55 PM

~~I thought it was a problem in master too because after testing your branch I pulled master and ran `make dev` and got the same (bad) result. But when I ran `make clean && make dev` on master it was good. This build target should be part of `make dev`? Oh, it's probably just the transition between this PR and master, once it's merged there won't be this build issue moving forward.~~ nvm

**Comment by ixrock** [comment]
@lensapp/lens-maintainers  ready for re-review / PTALA

Added 2 missing theme colours to `lens-dark.json`: `--layoutTabsLineColor`, `--canvasBackground` .

@jim-docker  light theme seems to be fixed for now:

image

File: src/renderer/components/+nodes/nodes.scss
Comment (jim-docker) [inline src/renderer/components/+nodes/nodes.scss:40]: Were `lensBlue` and `lensMagenta` defined somewhere?

**Review by jim-docker — APPROVED** [review]
LGTM

File: src/renderer/components/+nodes/nodes.scss
Comment (ixrock) [reply src/renderer/components/+nodes/nodes.scss:40]: Nope, they where defined in linking sass-vars to css-vars in deleted `theme-vars.scss`, e.g.

```scss
$lensBlue: var(--blue);
// ... other themed vars
```

**Pull request merged**
